Framework/DataHandling/src/SinglePeriodLoadMuonStrategy.cpp

```diff
@@ -0,0 +1,185 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidDataHandling/SinglePeriodLoadMuonStrategy.h"
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/GroupingLoader.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidDataHandling/ISISRunLogs.h"
+#include "MantidDataHandling/LoadMuonLog.h"
+#include "MantidDataHandling/LoadMuonNexusV2Helper.h"
+#include "MantidKernel/UnitFactory.h"
+#include "MantidKernel/UnitLabelTypes.h"
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace API;
+using namespace NeXus;
+using namespace HistogramData;
+using std::size_t;
+using namespace DataObjects;
+
+// Constructor
+SinglePeriodLoadMuonStrategy::SinglePeriodLoadMuonStrategy(
+    Kernel::Logger &g_log, const std::string &filename, NXEntry &entry,
+    Workspace2D_sptr workspace, int entryNumber, bool isFileMultiPeriod)
+    : LoadMuonStrategy(g_log, filename), m_entry(entry), m_workspace(workspace),
```

Comment:
It's worth doing `m_workspace(std::move(workspace))` to avoid incrementing the `shared_ptr` ref count twice.